models/meshmodel/core/v1alpha1/component.go

```diff
@@ -80,13 +80,24 @@ func CreateComponent(db *database.Handler, c ComponentDefinition) (uuid.UUID, er
 	err = db.Create(&cdb).Error
 	return c.ID, err
 }
-func GetMeshModelComponents(db *database.Handler, f ComponentFilter) (c []ComponentDefinition, count int64) {
+func GetMeshModelComponents(db *database.Handler, f ComponentFilter) (c []ComponentDefinition, count int64, unique int) {
 	type componentDefinitionWithModel struct {
 		ComponentDefinitionDB
 		ModelDB
 		CategoryDB
 	}
 
+	countUniqueComponents := func(components []componentDefinitionWithModel) int {
+		var set map[string]struct{}
+		for _, model := range components {
+			key := model.ComponentDefinitionDB.Kind + "@" + model.ModelDB.Name
```

Comment:
Is this how duplicity is checked, @harkiratsm? With a composite of these three items, @MUzairS15?

Reply:
@abdullah1308 PR doesn't have link to issue where can I see issue?

Reply:
@leecalcote [meshmap reference](https://github.com/layer5labs/meshery-extensions/blob/37bd8a7e82e27c133b1cb4cce238cadcac44cbdd/meshmap/src/utils/hooks/useMeshModelSearchModels.js#L43-L65)

Reply:
lint fixed